src/signup/signup.controller.js

```diff
@@ -0,0 +1,31 @@
+const validator = require('./signup.validator');
+const { getRequestBody } = require('../main.helpers');
+const { writeUser } = require('./signup.hellpers');
+
+class SignUpController {
+
+  constructor() { }
+  get signUpUser() {
+    return this._signUpUser.bind(this);
+  };
+
+  async _signUpUser(req, res) {
+    try {
+      const body = await getRequestBody(req);
+      const data = await validator.validateSignUpUser(body);
+      const user = await writeUser(data, res);
```

Comment:
It would be nice to check if such file already exist. If yes - then throw error to user (409).

Reply:
added check and error